web/packages/teleterm/src/services/tshd/grpcContextBridgeClient.ts

```diff
@@ -124,3 +126,29 @@ export function objectifyClient<T>(client: ServiceInfo): T {
     return previousValue;
   }, {} as T);
 }
+
+export type ObjectifiedAbortSignal = AbortSignal & {
+  canBePassedThroughContextBridge: true;
+};
+
+export function objectifyAbortSignal(a: AbortSignal): ObjectifiedAbortSignal {
```

Comment:
Two things:

1. `objectify` & `Objectified` as names are so-so. They describe what we do with the subject but not what for. I think `Cloneable` might work better here? Since we want to make the client and other entities cloneable so that they can pass the context bridge.
   * [The docs for context bridge](https://www.electronjs.org/docs/latest/api/context-bridge) uses the word "copied", but "Copyable" sounds worse IMHO than "Cloneable" and those object still technically count as "cloneable", as long as we're talking about [the structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm), which the docs for context bridge also mention.
2. We can use [a plain old JS constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new#description) to make the API a bit nicer! Well, that's just my opinion. But we don't have to use a verb for the name, just a noun in that case.
   * Those old-school constructors are not as common today since JavaScript now has proper support for classes, but they 1) still work 2) return objects and not classes.

Not sure if this is going to work in practice, but take a look:

```typescript
export type CloneableAbortSignal = ObjectifiedAbortSignal;

export function CloneableAbortSignal(
  this: CloneableAbortSignal,
  abortSignalOrController: AbortSignal | AbortController
): void {
  let a: AbortSignal;
  if (abortSignalOrController instanceof AbortController) {
    a = abortSignalOrController.signal;
  } else {
    a = abortSignalOrController;
  }

  this.canBePassedThroughContextBridge = true;
  this.onabort = (...args) => a.onabort(...args);
  this.throwIfAborted = () => a.throwIfAborted();
  this.dispatchEvent = (...args) => a.dispatchEvent(...args);
  this.addEventListener = (type, listener, options) =>
    a.addEventListener(type, listener, options);
  this.removeEventListener = (type, listener, options) =>
    a.removeEventListener(type, listener, options);
  this.eventListeners = (...args) => a.eventListeners(...args);
  this.removeAllListeners = (...args) => a.removeAllListeners(...args);
  // TODO: getters for reason and aborted
}
```

This API also allows passing either an abort signal or a controller, depending on which one is more convenient in a given situation.

Reply:
1. Agree, I'm not a fan of `objectify`, `cloneable` sounds much better.
2. I think I have never written a constructor function, but the idea sounds really nice. Unfortunately, it turned out that [TS doesn't support such functions](https://github.com/Microsoft/TypeScript/issues/2310), so result of `new CloneableAbortSignal()` is `any` :( 
I think it disqualifies this option 😓 
I'm considering renaming `objectifyAbortSignal` to `makeClonableAbortSignal` or `cloneAbortSignal` (but this one has slightly different meaning). WDYT?

Reply:
Ah damn, when I was writing this constructor, at first it didn't type check, so I changed it a little. Since the type check on it was passing, I assumed that it's all fine, but I forgot to check an actual usage of it. 🤦

`cloneAbortSignal` sounds fine, we are de facto "cloning" it into a new object after all.